baseplate.go

```diff
@@ -77,17 +77,42 @@ type Server interface {
 	Serve() error
 }
 
-// Serve runs the given Server until it is given an external shutdown signal
-// using runtimebp.HandleShutdown to handle the signal and shut down the
-// server gracefully.  Returns the (possibly nil) error returned by "Close" or
+// ServeArgs provides a list of arguments to Serve.
+//    server: Mandatory. Indicates the Server that should be run until
+//            receiving a shutdown signal.
+//    preShutdown: Optional. Indicates a slice of io.Closers that should
+//						be gracefully shut down before the server upon receipt of a
+//            shutdown signal.
+//    preShutdown: Optional. Indicates a slice of io.Closers that should
+//						be gracefully shut down after the server upon receipt of a
+//            shutdown signal.
+type ServeArgs struct {
+	server       Server
+	preShutdown  []io.Closer
+	postShutdown []io.Closer
+}
+
+// Serve runs the given Server until it is given an external shutdown signal.
+//
+// It uses runtimebp.HandleShutdown to handle the signal and gracefully shut
+// down, in order:
+//     * any provided preShutdown closers,
+//     * the server, and
+//     * any provided postShutdown closers.
+//
+// Returns the (possibly nil) error returned by "Close", or
 // context.DeadlineExceeded if it times out.
 //
 // If a StopTimeout is configure, Serve will wait for that duration for the
 // server to stop before timing out and returning to force a shutdown.
 //
 // This is the recommended way to run a Baseplate Server rather than calling
 // server.Start/Stop directly.
-func Serve(ctx context.Context, server Server) error {
+func Serve(ctx context.Context, serveArgs ServeArgs) error {
```

Comment:
style nit, calling this "serveArgs" is redundant, I think just "args" is simpler.